geode-core/src/main/java/org/apache/geode/internal/cache/wan/serial/BatchDestroyOperation.java

```diff
@@ -124,6 +124,19 @@ protected boolean operateOnRegion(CacheEvent event, ClusterDistributionManager d
             }
           }
         }
+
+        // destroy dropped event from unprocessedKeys
+        if ((Long) this.key == 0L && this.tailKey == -1) {
+          SerialGatewaySenderEventProcessor ep =
+              (SerialGatewaySenderEventProcessor) rgn.getSerialGatewaySender().getEventProcessor();
+          boolean removed = ep.basicHandlePrimaryDestroy(ev.getEventId());
```

Comment:
This will only destroy from unprocessedMap the secondary?  If we become primary in the meantime, that would be ok?

Reply:
yes